osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets;
+using osuTK;
+using osu.Game.Rulesets.UI;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using osuTK.Graphics;
+using System;
+using System.Linq;
+using osu.Framework.Extensions.IEnumerableExtensions;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class LeaderboardModSelector : CompositeDrawable
+    {
+        public readonly BindableList<Mod> SelectedMods = new BindableList<Mod>();
+        public readonly Bindable<RulesetInfo> Ruleset = new Bindable<RulesetInfo>();
+
+        private readonly FillFlowContainer<ModButton> modsContainer;
+
+        public LeaderboardModSelector()
+        {
+            AutoSizeAxes = Axes.Both;
+            InternalChild = modsContainer = new FillFlowContainer<ModButton>
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Full,
+                Spacing = new Vector2(4),
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Ruleset.BindValueChanged(onRulesetChanged, true);
+        }
+
+        private void onRulesetChanged(ValueChangedEvent<RulesetInfo> ruleset)
+        {
+            SelectedMods.Clear();
+            modsContainer.Clear();
+
+            if (ruleset.NewValue == null)
+                return;
+
+            modsContainer.Add(new ModButton(new ModNoMod()));
+            modsContainer.AddRange(ruleset.NewValue.CreateInstance().GetAllMods().Where(m => m.Ranked).Select(m => new ModButton(m)));
+
+            modsContainer.ForEach(button => button.OnSelectionChanged = selectionChanged);
+        }
+
+        private void selectionChanged(Mod mod, bool selected)
+        {
+            if (selected)
+                SelectedMods.Add(mod);
+            else
+                SelectedMods.Remove(mod);
+
+            updateHighlighted();
+        }
+
+        private void updateHighlighted()
+        {
+            if (SelectedMods.Any())
+                return;
+
+            modsContainer.Children.Where(button => !button.IsHovered).ForEach(button => button.IsActive.Value = !IsHovered);
```

Comment:
this makes less sense than ever. why is `IsActive= !IsHovered`

Reply:
Let's assume everything is not selected, so all the buttons are active. On hover the container itself we want every button to became gray (`button.IsActive = false`), and on hover lost we want every button to became normal (`button.IsActive = true`). That means, `button.IsActive` depends on hovering the container (except cases if we have selected buttons)

Reply:
Naming would go a long way here at the least. If you had named those properties `MouseOverContainer` instead of `IsActive` and `MouseOverButton` instead of `IsHovered` or something of the sort this would be way less confusing to read. I kind of internalized the meaning of those names when I was reading this source but I should have pointed out that they're not very helpful.

Reply:
Well, `IsActive` don't always means it's hovered and `IsHovered` is a default `Drawable` property

Reply:
why not call it "Dimmed"?

Reply:
I'm renaming almost every property atm. So `Highlighted` for mod icon is now `Selected` and `IsActive` is `Highlighted`. It makes more sense now and easier to understand what's going on